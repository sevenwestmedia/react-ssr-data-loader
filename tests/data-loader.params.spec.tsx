import { ComponentWithArgsFixture } from './helpers/component-with-args-fixture'
import Adapter from 'enzyme-adapter-react-16'
import { configure } from 'enzyme'
import { processEventLoop } from './helpers/event-loop-helpers'
import { act } from 'react-dom/test-utils'

configure({ adapter: new Adapter() })

describe('data-loader', () => {
    it('can specify arguments for data loader', async () => {
        const args = { bar: 1, id: 'testKey', resourceType: 'testDataType' }
        const sut = new ComponentWithArgsFixture(undefined, 'testKey', args, undefined, false)

        await act(() => sut.testDataPromise.resolve({ result: 'Test' }))

        expect(sut.passedParams).toEqual({ ...args, paramsCacheKey: '7a4496e0' })
        expect(sut.resources.generateCacheKey('testDataType', args)).toBe('7a4496e0')
        const keys = Object.keys(sut.getLoaderState() || {})
        expect(keys).toContain('7a4496e0')
        sut.assertState()
    })

    it('can control which params are used in cache key', async () => {
        const args = { bar: 1, id: 'testKey', resourceType: 'testDataType' }
        const sut = new ComponentWithArgsFixture(undefined, 'testKey', args, ['id'], false)

        await act(() => sut.testDataPromise.resolve({ result: 'Test' }))

        const state = { ...sut.currentState }
        act(() =>
            sut.setState({
                ...sut.getLoaderState(),
                bar: 2,
            }),
        )

        await processEventLoop()
        expect(state).toEqual(sut.currentState)
    })
})
