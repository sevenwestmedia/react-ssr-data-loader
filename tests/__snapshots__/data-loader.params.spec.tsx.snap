// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data-loader can specify arguments for data loader 1`] = `
Object {
  "loadAllCompletedCalled": 1,
  "loadDataCount": 1,
  "passedParams": Object {
    "bar": 1,
    "id": "testKey",
    "paramsCacheKey": "7a4496e0",
    "resourceType": "testDataType",
  },
  "renderActions": Object {
    "refresh": [Function],
  },
  "renderCount": 2,
  "renderProps": Object {
    "data": Object {
      "dataFromServerSideRender": false,
      "hasData": true,
      "result": Object {
        "result": "Test",
      },
    },
    "lastAction": Object {
      "success": true,
      "type": "fetch",
    },
    "status": "Idle",
  },
}
`;
